lib/index.js

```diff
@@ -286,7 +286,7 @@ function createLocation(ctx, node, location) {
       let key
 
       for (key in location.attrs) {
-        if (own.call(location.attrs, key)) {
+        if (location.attrs && own.call(location.attrs, key)) {
```

Comment:
There is no good reason for this code inside the loop (other than TypeScript), it looks like a bug that a constant condition is checked on every iteration.

If TypeScript is indeed correct that `location.attrs` can be missing, then we already have an if-branch where that could be checked: L282. What do you think of using that?